src/workerd/io/worker.h

```diff
@@ -334,6 +334,24 @@ class Worker::Isolate: public kj::AtomicRefcounted {
 
   kj::Own<const WeakIsolateRef> getWeakRef() const;
 
+  struct MemStats {
```

Comment:
Why is this type declared separately in `Worker` and `jsg::IsolateBase`? Can both places share one type declaration?

In fact, could both just use `v8::HeapStatistics` directly instead of defining our own struct(s)? These stats are very specific to V8 anyway; it's not like a different JS engine is going to export the same statistics. So defining our own parallel structs seems to just add boilerplate IMO.